packages/compass-aggregations/src/components/pipeline-results-workspace/index.tsx

```diff
@@ -1,24 +1,134 @@
 import React from 'react';
 import { connect } from 'react-redux';
-import type { ConnectedProps } from 'react-redux';
+import type { Document } from 'mongodb';
+import {
+  IconButton,
+  css,
+  spacing,
+  Icon,
+  Button,
+  Body,
+  uiColors,
+  cx,
+} from '@mongodb-js/compass-components';
 
 import type { RootState } from '../../modules';
+import {
+  fetchNextPage,
+  fetchPrevPage,
+  cancelAggregation,
+} from '../../modules/aggregation';
+
+type PipelineResultsWorkspace = {
+  documents: Document[];
+  page: number;
+  perPage: number;
+  loading: boolean;
+  isPrevDisabled: boolean;
+  isNextDisabled: boolean;
+  error?: string;
+  onPrev: () => void;
+  onNext: () => void;
+  onCancel: () => void;
+};
+
+const topStyles = css({
+  display: 'flex',
+  justifyContent: 'flex-end',
+  alignItems: 'center',
+  gap: spacing[2],
+});
+
+const centeredContentStyles = css({
+  textAlign: 'center',
+  marginTop: spacing[4],
+});
+
+const errorMessageStyles = css({
+  color: uiColors.red.base,
+});
 
 const PipelineResultsWorkspace: React.FunctionComponent<PipelineResultsWorkspace> =
-  ({ documents }) => {
+  ({
+    documents,
+    page,
+    perPage,
+    loading,
+    isPrevDisabled,
+    isNextDisabled,
+    error,
+    onPrev,
+    onNext,
+    onCancel,
+  }) => {
+    const showingFrom = (page - 1) * perPage;
+    const showingTo = showingFrom + documents.length;
     return (
       <div data-testid="pipeline-results-workspace">
-        <pre>
-          <code>{JSON.stringify(documents, null, 2)}</code>
-        </pre>
+        <div className={topStyles}>
+          <Body>
+            Showing {showingFrom + 1} - {showingTo}
+            &nbsp;of count.
+          </Body>
+          <div>
+            <IconButton
+              aria-label="Previous page"
+              disabled={isPrevDisabled || loading}
+              onClick={() => onPrev()}
+            >
+              <Icon glyph="ChevronLeft" />
+            </IconButton>
+            <IconButton
+              aria-label="Next page"
+              disabled={isNextDisabled || loading}
+              onClick={() => onNext()}
+            >
+              <Icon glyph="ChevronRight" />
+            </IconButton>
+          </div>
+        </div>
+        {error && (
+          <Body className={cx(centeredContentStyles, errorMessageStyles)}>
+            {error}
+          </Body>
+        )}
+        {loading && (
+          <div className={centeredContentStyles}>
+            <Body>Loading ... </Body>
+            <Button
+              onClick={() => onCancel()}
+              variant="primaryOutline"
+              size="xsmall"
+            >
+              Cancel
+            </Button>
+          </div>
+        )}
+        {documents.length > 0 && (
+          <pre>
+            <code>{JSON.stringify(documents, null, 2)}</code>
```

Comment:
Without being entirely sure about the context here, should this be EJSON rather than JSON? (I assume it’s not the final state of things because we’d probably use one of our existing components with code hightlighting etc. here anyway, so feel free to just resolve this if I’m being unhelpful)